src/traces/bar/cross_trace_calc.js

```diff
@@ -119,8 +123,35 @@ function setGroupPositions(gd, pa, sa, calcTraces, opts) {
             }
             break;
     }
+    collectExtents(calcTraces, pa, opts);
+}
+
+function standardizeCornerradius(calcTraces) {
+    if(!calcTraces.length) return;
+    var i, calcTrace, fullTrace;
+    var firstTrace = calcTraces[0][0].trace;
+    // If first trace has cornerradius, set all other traces to same cornerradius
+    if(firstTrace.marker && firstTrace.marker.cornerradius) {
```

Comment:
Good point, for consistency I guess we would want to explicitly set cornerradius of the other traces to 0 for consistency with the rest of the logic. So this `if` should check whether `cornerradius` is _defined_ instead of checking whether it's truthy.

Reply:
@alexcjohnson @archmoj FWIW I'm not totally happy with this logic, from a UX perspective I would rather check whether _any_ trace in the group has a cornerradius and if so, set all traces in the group to that cornerradius, rather than just going by the first trace.

But, it seems too slow to iterate through every single trace to check, since this code will run for every bar chart, cornerradius or not. So this is the compromise I arrived at. Open to suggestions.

Reply:
I think it should be `defined` and it should be a valid value.
I suggest you possibly move the validation part to supply default step.
So if the value is not a valid input then we set it to *undefined* right after `coerce`.

Reply:
I'm not even sure about constraining it to only visible traces - seems like that could give strange experiences when you hide/show via legend. maybe all visible truthy (ie `true` or `'legendonly'`) traces on the same subplot

Reply:
> We should loop through `bar` & `histogram` traces and break the loop with the first visible one that has a `cornerradius` as you also mentioned. No?

Yep, makes sense to me as long as you think that's OK from a performance perspective.

Reply:
> I'm not even sure about constraining it to only visible traces - seems like that could give strange experiences when you hide/show via legend. maybe all visible truthy (ie true or 'legendonly') traces on the same subplot

@alexj Unfortunately I've realized `"legendonly"` traces are excluded earlier (on line 32 in `cross-trace-calc.js`) so including them is non-trivial — will leave as-is unless you feel strongly they should be included.